src/baitapquanlysinhvien/SinhVien.java

```diff
@@ -0,0 +1,198 @@
+import java.util.Comparator;
+import java.util.Scanner;
+
+public class SinhVien {
+	private int maSV;
+	private String hoTen;
+	private LocalDate ngayThang;
+	private int namNhaphoc;
+	private double diemDauvao;
+	ArrayList<KetQuaHocTap> kq = new ArrayList<KetQuaHocTap>();
+
+	public SinhVien(int maSV, String hoTen, LocalDate ngayThang, int namNhaphoc, double diemDauvao,
+			ArrayList<KetQuaHocTap> kq) {
+		this.maSV = maSV;
+		this.hoTen = hoTen;
+		this.ngayThang = ngayThang;
+		this.namNhaphoc = namNhaphoc;
+		this.diemDauvao = diemDauvao;
+		this.kq = kq;
+	}
+
+	public int getMaSV() {
+		return maSV;
+	}
+
+	public void setMaSV(int maSV) {
+		this.maSV = maSV;
+	}
+
+	public String getHoTen() {
+		return hoTen;
+	}
+
+	public void setHoTen(String hoTen) {
+		this.hoTen = hoTen;
+	}
+
+	public LocalDate getNgayThang() {
+		return ngayThang;
+	}
+
+	public void setNgayThang(LocalDate ngayThang) {
+		this.ngayThang = ngayThang;
+	}
+
+	public int getNamNhaphoc() {
+		return namNhaphoc;
+	}
+
+	public void setNamNhaphoc(int namNhaphoc) {
+		this.namNhaphoc = namNhaphoc;
+	}
+
+	public double getDiemDauvao() {
+		return diemDauvao;
+	}
+
+	public void setDiemDauvao(double diemDauvao) {
+		this.diemDauvao = diemDauvao;
+	}
+
+	public ArrayList<KetQuaHocTap> getKq() {
+		return kq;
+	}
+
+	public void setKq(ArrayList<KetQuaHocTap> kq) {
+		this.kq = kq;
+	}
+
+	public void hienThithongtin() {
+		System.out.println("SinhVien [maSV=" + maSV + ", hoTen=" + hoTen + ", ngayThang=" + ngayThang + ", namNhaphoc="
+				+ namNhaphoc + ", diemDauvao=" + diemDauvao + ", kq=" + kq + "]");
+	}
+
+	public static void hienThiSVChinhQuy(ArrayList<SinhVien> dsSV) {
+		System.out.println("Danh sách sinh viên chính quy là: ");
+		for (int i = 0; i < dsSV.size(); i++) {
+			SinhVien sv = dsSV.get(i);
+			if (sv instanceof ChinhQuy) {
+				sv.hienThithongtin();
+			}
+		}
+	}
+
+	public static void hienThidiemtrungbinh(ArrayList<SinhVien> dsSV) {
+		Scanner sc = new Scanner(System.in);
+		System.out.println("Nhập mã sinh viên bạn muốn hiển thị điểm trung bình: ");
+		int maSV = sc.nextInt();
+		for (int i = 0; i < dsSV.size(); i++) {
+			SinhVien sv = dsSV.get(i);
+			if (maSV == sv.maSV) {
+				System.out.println(sv.hoTen);
+				for (int j = 0; j < sv.kq.size(); j++) {
+					KetQuaHocTap kq = sv.kq.get(j);
+					kq.hienThiDiem();
+				}
+			}
+		}
+	}
+
+	public static void getSVtheoNoiLienKet(ArrayList<SinhVien> dsSV) {
+		ArrayList<TaiChuc> svTaiChuc = new ArrayList<TaiChuc>();
+		Scanner sc = new Scanner(System.in);
+		System.out.println("Nhập nơi liên kết bạn muốn tìm kiếm: ");
+		String noiLienKet = sc.nextLine();
+		for (int i = 0; i < dsSV.size(); i++) {
+			SinhVien sv = dsSV.get(i);
+			if (sv instanceof TaiChuc) {
+				svTaiChuc.add((TaiChuc) sv);
+			}
+		}
+		for (int i = 0; i < svTaiChuc.size(); i++) {
+			TaiChuc svtc = svTaiChuc.get(i);
+			if (svtc.getNoiLienKet().equals(noiLienKet)) {
+				svtc.hienThithongtin();
+			}
+		}
```

Comment:
thích xem code gọn hơn hơm :v 
```suggestion
		List<TaiChuc> svTaiChuc =  dsSV.stream().filter(student -> student instanceof TaiChuc).collect(Collectors.toList());
		List<TaiChuc> svTaiChucResult = svTaiChuc.stream().filter(student -> student.getNoiLienKet().equal(noiLienKet)).collect(Collectors.toList());
		svTaiChucResult.forEach(sv -> sv.hienThithongtin());
```